### PR TITLE
[15.0][MIG] website_require_login: Migration to 15.0 

### PR DESCRIPTION
**Pull request by NICO-SOLUTIONS**
Standard migration

### CODE REVIEW

**Review by Aldeigja — APPROVED** [review]
Functional ok

**Review by geomer198 — APPROVED** [review]
Code review ok!

**Comment by NICO-SOLUTIONS** [comment]
@OCA/website-maintainers could you merge please?

File: website_require_login/tests/test_ir_http.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
+from odoo.tests import HttpCase
+
+
+class TestIrHttp(HttpCase):
+    def setUp(self):
```
Comment (simahawk) [inline website_require_login/tests/test_ir_http.py]: Can you please move this code to `setUpClass`?
`setUp` runs for _every_ test which means that in this case creates and re-creates records all the times.
Not a big deal w/ 2 tests but if you multiply this for all the modules we have... :wink:

**Review by simahawk — COMMENTED** [review]
LG, minor remark

File: website_require_login/tests/test_ir_http.py (hunk 1 above)
Comment (NICO-SOLUTIONS) [reply website_require_login/tests/test_ir_http.py]: Thanks for the input. Done!
Will do the same for #1019

File: website_require_login/tests/test_ir_http.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        cls.user = cls.env["res.users"].create(
+            {"name": "Test User", "login": "test_user", "password": "12345"}
+        )
+        cls.path = "/contactus"
+        cls.expected_path = "/web/login?redirect=%s" % cls.path
+
+    def test_dispatch_unauthorized(self):
+        # Test that a public user cannot access "/auth_path
+        self.authenticate(None, None)
+        response = self.url_open(self.__class__.path, allow_redirects=False)
```
Comment (simahawk) [inline website_require_login/tests/test_ir_http.py]:
```suggestion
        response = self.url_open(self.path, allow_redirects=False)
```

File: website_require_login/tests/test_ir_http.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+        self.assertEqual(
+            response.status_code,
+            303,
+            "Expected the response status code to be 303 indicating a redirect",
+        )
+
+        self.assertIn(self.__class__.expected_path, response.headers["Location"])
```
Comment (simahawk) [inline website_require_login/tests/test_ir_http.py]:
```suggestion
        self.assertIn(self.expected_path, response.headers["Location"])
```

File: website_require_login/tests/test_ir_http.py [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+    def test_dispatch_authorized(self):
+        # Test that an authorized user can access "/auth_path
+        self.authenticate(user="test_user", password="12345")
+        response = self.url_open(self.__class__.path)
```
Comment (simahawk) [inline website_require_login/tests/test_ir_http.py]:
```suggestion
        response = self.url_open(self.path)
```

**Review by simahawk — APPROVED** [review]
You can squash the last commit on top to the migration one. If you keep it separated, just rewrite it to include the module name. Something like `[IMP] website_require_login: boost test setup`.
pre-approving

**Comment by NICO-SOLUTIONS** [comment]
@OCA/website-maintainers 
Can we merge this one?

**Comment by NICO-SOLUTIONS** [comment]
@OCA/website-maintainers 
can we remove the stale tag and/or merge this one?

**Comment by NICO-SOLUTIONS** [comment]
@hbrunn thx!

**Pull request merged**